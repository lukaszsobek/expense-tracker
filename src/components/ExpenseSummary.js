import React from "react"
import { connect } from "react-redux"
import numeral from "numeral"

import { getExpensesSummary, getVisibleExpenses } from "../selectors"

export const ExpenseSummary = ({ expenses }) => {
    const { count, total } = getExpensesSummary(expenses)
    const summaryCount = count === 1 ? "1 expense " : `${count} expenses `
    const summaryTotal = numeral(total / 100).format("$0,0.00")
    return (
        <div className="expense-summary">
            Viewing {summaryCount} totalling {summaryTotal}
        </div>
    )
}

const mapStateToProps = ({ expenses, filters }) => ({
    expenses: getVisibleExpenses(expenses, filters)
})

export default connect(mapStateToProps)(ExpenseSummary)

