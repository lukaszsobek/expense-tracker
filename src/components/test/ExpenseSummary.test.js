import React from "react"
import { ExpenseSummary } from "../ExpenseSummary"
import { shallow } from "enzyme"

describe("ExpenseSummary renders",() => {
    it("with 0 expenses",() => {
        const wrapper = shallow(<ExpenseSummary />)
        expect(wrapper).toMatchSnapshot()
    })

    it("with 1 expense",() => {
        const expenses = [
            {amount: 10}
        ]
        const wrapper = shallow(<ExpenseSummary expenses={expenses} />)
        expect(wrapper).toMatchSnapshot()
    })

    it("with 3 expenses",() => {
        const expenses = [
            {amount: 10},
            {amount: 20},
            {amount: 30}
        ]
        const wrapper = shallow(<ExpenseSummary expenses={expenses} />)
        expect(wrapper).toMatchSnapshot()
    })
})
