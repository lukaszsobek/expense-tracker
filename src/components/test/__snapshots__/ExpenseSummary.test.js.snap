// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExpenseSummary renders with 0 expenses 1`] = `
<div
  className="expense-summary"
>
  Viewing 
  0 expenses 
   totalling 
  $0.00
</div>
`;

exports[`ExpenseSummary renders with 1 expense 1`] = `
<div
  className="expense-summary"
>
  Viewing 
  1 expense 
   totalling 
  $0.10
</div>
`;

exports[`ExpenseSummary renders with 3 expenses 1`] = `
<div
  className="expense-summary"
>
  Viewing 
  3 expenses 
   totalling 
  $0.60
</div>
`;
